addons/binding/org.openhab.binding.tivo/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,152 @@
+control of channel changes, user interface navigation and allow the client to send simulated remote control button presses to the Digital Video Recorder.<![CDATA[ <br /><br /> ]]>See https://www.tivo.com/assets/images/abouttivo/resources/downloads/brochures/TiVo_TCP_Network_Remote_Control_Protocol.pdf for supported commands / codes for each channel.</description>
+
+        <channels>
+            <channel id="tivoChannelForce" typeId="tivoChannelForce"/>
+            <channel id="tivoChannelSet" typeId="tivoChannelSet"/>
+            <channel id="tivoTeleport" typeId="tivoTeleport"/>
+            <channel id="tivoIRCommand" typeId="tivoIRCommand"/>
+            <channel id="tivoKBDCommand" typeId="tivoKBDCommand"/>
+            <channel id="tivoStatus" typeId="tivoStatus"/>
+            <channel id="tivoCommand" typeId="tivoCommand"/>
+        </channels>
+        
+        <config-description>
+        	<parameter name="deviceName" type="text" required="true">
+				<default>My Tivo</default>
+				<required>true</required>
+				<label>Device Name</label>
+				<description>A friendly name to refer to this device.<![CDATA[ <b> ]]>Default: Device name specified on the TiVo.<![CDATA[ </b> ]]></description>
+			</parameter>
+        	<parameter name="address" type="text" required="true">
+            	<default></default>
+            	<required>true</required>
+            	<label>Address</label>
+            	<description>The IP address or host name of your TiVo box</description>
+        	</parameter>
+        	<parameter name="tcpPort" type="integer" max="65535" min="1">
+				<default>31339</default>
+				<required>true</required>
+				<label>TCP Port</label>
+				<context>network-address</context>
+				<description>The TCP port number used to connect to the TiVo.<![CDATA[ <b> ]]>Default: 31339<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="numRetry" type="integer" max="20" min="0">
+				<default>5</default>
+				<required>true</required>
+				<label>Connection Retries</label>
+				<description>The number of times to attempt reconnection to the TiVo box, if there is a connection failure.<![CDATA[ <b> ]]>Default: 5<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="keepConActive" type="boolean">
+				<default>true</default>
+				<required>true</required>
+				<label>Keep Connection Open</label>
+				<description>Keep connection to the TiVo open.  Recommended for monitoring the TiVo for changes in TV channels.  Disable if other applications that use the Remote Control Protocol port will also be used e.g. mobile remote control applications.<![CDATA[ <b> ]]>Default: True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollForChanges" type="boolean">
+				<default>true</default>
```

Comment:
okay, so the Tivo will never give you updates by itself even when the connection is kept alive. You will always have to poll the status if you want to keep up with commands issued from another device. tnx for clarification.